### ISSUE TITLE
Refactor Swift storage driver properties / properties getter

### ISSUE THREAD

**Issue by carlosmcgregor** [IQSS/dataverse#5478]
The Swift storage driver loads properties from `swift.properties` with every action executed by the driver (e.g. [when creating a temp url](https://github.com/IQSS/dataverse/blob/develop/src/main/java/edu/harvard/iq/dataverse/dataaccess/SwiftAccessIO.java#L770)). The `getter` itself, which loads the file, is located [in these lines](https://github.com/IQSS/dataverse/blob/develop/src/main/java/edu/harvard/iq/dataverse/dataaccess/SwiftAccessIO.java#L687-L696).

Loading properties would be better suited for a `class` like (`SystemConfig.java`)[https://github.com/IQSS/dataverse/blob/develop/src/main/java/edu/harvard/iq/dataverse/util/SystemConfig.java]. Please correct me if this would not be the best place!

This also opens a bigger argument on whether `swift.properties` should be on a separate file than `domain.xml`. If not, then we could forgo the previously mentioned refactoring of the `getter` and move ahead with loading the configurations with `System.getProperty()` ([à la S3 driver](https://github.com/IQSS/dataverse/blob/develop/src/main/java/edu/harvard/iq/dataverse/dataaccess/S3AccessIO.java#L109)).

**Comment by pdurbin** [comment]
@carlosmcgregor thank you for popping into IRC the other day to discuss this idea ( http://irclog.iq.harvard.edu/dataverse/2019-01-11#i_84497 ) and for creating this issue. Since you've already made improvements to the Swift driver in pull request #5356 you're in a better position than I am to judge how to make additional improvements, such as the performance optimizations you're proposing. The options as I understand them are:

- Status quo: leave the config in swift.properties
- Move config to JVM options (domain.xml)
- Move the config to database settings

Are you interested in creating a pull request? Do you you want some more opinions from other Dataverse developers, first? If so, I can pull this issue into the code review column at https://waffle.io/IQSS/dataverse to bring attention to it during our daily standup. Please let us know. Thanks!

**Comment by carlosmcgregor** [comment]
@pdurbin for "status quo", I was thinking about retaining `swift.properties` but moving the properties getter somewhere else so that the file is not accessed on every transaction.

**Comment by pdurbin** [comment]
Oh! I understand, I think. A behind-the-scenes optimization. Maybe the doc change would mention that if you need to update `swift.properties` you have to restart Glassfish? It sounds like right now the moment you change the file the new settings are live.

**Comment by carlosmcgregor** [comment]
My apologies, @pdurbin ~! I studied the issue further and noticed such optimization would not be possible whilst keeping a `swift.properties` file. Hence, we could load all properties into `domain.xml`. 

If credentials are a security concern, we could keep all credential data in a `swift.properties` file, thus giving flexibility if you were to, at one point, secure that file in some other place (similar to what the S3 driver does standard, having all credentials located at `~/`), and move all other regular settings like Swift URL to `domain.xml`. This would also allow for `asadmin` changes to these values and create greater parallelism between the way S3 and Swift is administered in Dataverse.

**Comment by pdurbin** [comment]
@carlosmcgregor for storage of credentials, please take a look a the "create-password-alias" work that was recently merged in pull request #5487. Thanks!

**Comment by pdurbin** [comment]
@carlosmcgregor thank you very much for pull request #5680 !

@poikilotherm and I just chatted about it and http://irclog.iq.harvard.edu/dataverse/2019-03-25#i_88874 and he has a preference for dots and camelCase instead of dashes.

So, can you please change (for example)...

    dataverse.file.swift-auth-type

... to ...

    dataverse.file.swift.authType

Thanks!

Please see also https://github.com/IQSS/dataverse-kubernetes/issues/11 about how dashes are not allowed in environment variables.

**Comment by carlosmcgregor** [comment]
Thank you @pdurbin and @poikilotherm ! I will be making the changes.

**Comment by pdurbin** [comment]
@landreev not sure if you've seen the long list of other JVM options with dashes that are on the (future) chopping block: https://github.com/IQSS/dataverse-kubernetes/issues/11#issuecomment-477335158

**Comment by scolapasta** [comment]
@carlosmcgregor thanks for the PR. We're planning in discussing this issue and the approach next week at our weekly tech hours.

**Comment by poikilotherm** [comment]
I really appreciate the work of @carlosmcgregor - it inspired me to open #5733.

Let me wish for something in addition to this great work: I would really love to see a small wrapper around the property retrieval which later on allows usage of aliases. Aliases would be needed for attacking #5733 and others from IQSS/dataverse-kubernetes#11 to ensure backward compatibility.

If this should be done in a separate issue/step, just ignore me.

**Comment by pdurbin** [comment]
At standup we heard that we got the Joss developers' attention. We sure did because now @larsks in #dataverse IRC and just opened #5750.

I think we can unblock this issue (#5478) by writing a little Java code for #5750.

**Comment by pdurbin** [comment]
@carlosmcgregor is your code backward compatible for existing Swift installations? Can you please add a release notes doc? The procedure for how to add a release notes doc is explained here: https://github.com/IQSS/dataverse/tree/v4.12/doc/release-notes#release-notes-directory-process . Thanks!

**Comment by carlosmcgregor** [comment]
Hello @pdurbin ! I will address the changes administrators have to do to the configuration files in the release notes.

**Comment by pdurbin** [comment]
@carlosmcgregor thanks! At standup just now we were thinking that you hadn't yet merged the latest from develop but I see you did this already last night. This will unblock QA so thank you!

Please drop a note in here when you're done with the release notes. Thanks!

**Comment by carlosmcgregor** [comment]
@pdurbin Sorry~! Just reading the release notes doc I noticed that I did not branch my fork w/ the issue number. Should I close this PR, branch, and make another PR addressing this PR?

**Comment by pdurbin** [comment]
@carlosmcgregor no, please don't worry about the name of your branch. Putting the number in there is just a suggestion. You can put the number in the release notes file, though. "5478-swift.md" or whatever. Thanks.

**Comment by carlosmcgregor** [comment]
@pdurbin Thank you and done!
